pkg/util/loop/loop.go

```diff
@@ -115,6 +143,9 @@ func (loop *Device) AttachFromFile(image *os.File, mode int, number *int) error
 					continue
 				}
 			}
+			if eagainFound {
+				return &TransientAttachError{"failed to successfully allocate a loop device (please retry)"}
+			}
```

Comment:
Ok, I guess this makes sense, though this code could equally be after the loop rather than awkwardly jammed in here at the top of the loop (I think the existing conditional starting at L138 `if device == loop.MaxLoopDevices` is an example of fairly poor practice, and I am loath to join it). 

I guess your thinking here is probably that the existing code has a block here to handle reaching the end of the loop, so it makes sense to add code that we need to handle the end of the loop here. I think that does make some sense.

However, in this case in order to fit in with this model we must add an extra tracking variable `eagainFound` that would not be necessary if we refactor our code into the post-loop block. 

The primary reason the existing conditional block starting at L138 exists is, in my opinion, because they want to be able to give up on shared mode and override the `device` loop variable with one they recently found and then continue with the rest of the loop code that will (hopefully) assign that device. I don't think I could work with this code without refactoring out the dangers inherent in overriding the loop variable from inside the loop. That is, in my experience, almost always a terrible idea. 

The way to refactor this would be to pull out the remaining part of the per-device loop into a separate function that takes `device` as an argument, change the loop condition to be `device < loop.MaxLoopDevices`, and add a conditional block after the loop that handles `if loop.Shared` and `if freeDevice` by invoking that inner function once more. 

Following those conditionals, the final return value for the function can safely be to simply `return &TransientAttachError{"failed to successfully allocate a loop device (please retry)"}` without any need to check (or track) `eagainFound`.